scripts/SetupServiceFabric/SetupServiceFabric.sh

```diff
@@ -26,6 +31,122 @@ if [[ "xenial" != "$Distribution" && "bionic" != "$Distribution" ]]; then
     exit -1
 fi
 
+# Check ServiceFabricRuntimePath and ServiceFabricSdkPath have been provided. If yes, extract that
+ServiceFabricRuntimePath=""
+ServiceFabricSdkPath=""
+for i in "$@"
+do
+case $i in
+    -sfrt=*|--servicefabricruntime=*)
+    ServiceFabricRuntimePath="${i#*=}"
+    ;;
+    -sfs=*|--servicefabricsdk=*)
+    ServiceFabricSdkPath="${i#*=}"
+    ;;
+    *)
+    echo Error: Unknown option passed in: $i
+    exit 1
+    ;;
+esac
+done
+
+# Check if systemd is running as PID1, if not it should be enabled via system-genie
+pidone=$(ps --no-headers -o comm 1)
+if [ "systemd" != "$pidone" ]; then
```

Comment:
what will happen when this script is not running on WSL2 but on a real Linux system ?

Reply:
When this script is running on a real linux system, it will not enter into genie code block and installation will happen as it was happening earlier.